### ISSUE TITLE
Add a setting to control selection foreground color

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#3580]
This is highly related to #3326/#3471 and #3561.

We've added a setting to control what color we use to highlight text with when it's selected in #3471.

However, that doesn't _necessarily_ fix our contrast ratio problems w.r.t. selection, since the configured color might still not have enough difference from the foreground of the highlighted text.

Conhost had an inverting selection, which didn't have this problem. However, we think that might be Hard to implement in the dx renderer. Adding that to the Terminal is being tracked in #3561. 

A selection foreground color 
A: might be more feasible to implement
B: is a good configurable setting to add regardless, to enable further personalization of the terminal.

Implementation might still be tricky though, since the Renderer is only ever told about the selection rects _after_ it has drawn the text. Re-drawing the text seems like a bad idea. I'm not sure we'll be able to re-order things in the Renderer safely without also impacting the GDI and other rendering heads. This is hard, but easier than inverting selection.

**Comment by mdtauk** [comment]
I suggested this in the Selection Background colour issue, so I am all for this.

**Comment by j4james** [comment]
Note that both XTerm and VTE support OSC 17 and 19 for settings these colors, so that's something we may also want to support in the future. Not essential, but I thought it worth mentioning now in case it's something you need to take into account in the design.

**Comment by zadjii-msft** [comment]
@j4james I did _not_ know that. That's definitely good info to have, since I definitely said during the #3471 PR "this doesn't need to be in the terminal core, it's not something that's set by VT" :egg_on_face:

It's not a terrible pain to refactor the existing code to support that. We'll make sure to keep that in mind when coming back to this ☺️

**Comment by javierdlg** [comment]
My team is very interested in this as well! Being able to set the background selection text color fixes half of the problem, and setting the foreground as well would clear up our contrast ration issues.

**Comment by javierdlg** [comment]
@zadjii-msft, has this issue been slated for a release?

**Comment by DHowett-MSFT** [comment]
@javierdlg no. when it is slated for a release, the _Milestone_ on the righthand side will change from "Backlog". :smile:

**Comment by rub8n** [comment]
@DHowett-MSFT not having this setting is impacting our accessibility story (e.g. high contrast themes) and thus are ability to release our terminal in the release channel. 

Do you have any estimates on when you'll be able to expose this setting?

**Comment by DHowett-MSFT** [comment]
This is a significant amount of engineering work (cf. the original post) that I can't commit to before Terminal v1.0. I have, however, marked it Help-Wanted. :smile:

**Comment by nnathan** [comment]
Any update on this? It's been a few years, for accessibility reasons would love to be able to change the selection foreground colour.
